stratagemScripts/examples.strata

```diff
@@ -1,4 +1,4 @@
 let x: Int = 0 in x + 1;
 let x: Int = 3 + 4 in x;
 let y: Bool = true in ();
-fn(z: Bool): Bool { z; z }(false)
+fn(z: Bool): Bool { z; z }(let b: Bool = true in b)
```

Comment:
This would have crashed before with the following error:

```
Exception in thread "main" edu.sjsu.stratagem.exception.StratagemTypecheckException: Function's body doesn't have ascribed type, ascribed: null, had: Bool
	at edu.sjsu.stratagem.FunctionDeclExpr.typecheck(Expression.java:198)
	at edu.sjsu.stratagem.FunctionAppExpr.typecheck(Expression.java:133)
	at edu.sjsu.stratagem.FunctionAppExpr.typecheck(Expression.java:142)
	at edu.sjsu.stratagem.SeqExpr.typecheck(Expression.java:294)
	at edu.sjsu.stratagem.Interpreter.main(Interpreter.java:34)
```

because we were constructing let-expressions as having a `null` return type. They were basically in limbo until we either introduced type inference (as per this PR) or explicit type annotating (as described in #4).